components/compositing/largest_contentful_paint.rs

```diff
@@ -0,0 +1,91 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+use base::cross_process_instant::CrossProcessInstant;
+use compositing_traits::largest_contentful_paint_candidate::{
+    LCPCandidate, LargestContentfulPaint,
+};
+use fnv::{FnvBuildHasher, FnvHashMap};
+use webrender_api::{Epoch, PipelineId};
+
+/// Holds the [`LargestContentfulPaintCalculator`] for each pipeline.
+#[derive(Default)]
+pub(crate) struct LargestContentfulPaintDetector {
+    lcp_calculators: FnvHashMap<PipelineId, LargestContentfulPaintCalculator>,
+}
+
+impl LargestContentfulPaintDetector {
+    pub const fn new() -> Self {
```

Comment:
The default constructor should better be implemented via the `Default` trait instead of `new()`